server/queries/db.js

```diff
@@ -17,23 +18,98 @@ sequelize.sync({ alter: true }).then(() => {
     console.log("Database connected.");
 });
 
-module.exports.createUser = username => {
+module.exports.validatePassword = (username, password) => {
+    return new Promise((resolve, reject) => {
+        User.findOne({
+            where: {
+                username: username
+            }
+        })
+        .then(user => {
+            if (user) {
+                bcrypt.compare(password, user.dataValues.password, (err, res) => {                    
+                    if (res) {
+                        resolve(user);
+                    } else {
+                        reject({
+                            code: 404,
+                            message: "Password is incorrect"
+                        });
+                    }
+                });
+            } else {
+                reject({
+                    code: 404,
+                    message: "User with this username does not exist"
```

Comment:
sometimes it's considered bad to reveal whether a user exists. you have to decide whether you would prefer usability, or privacy+security.

Reply:
usability